src/phpDocumentor/Descriptor/Collection.php

```diff
@@ -149,6 +154,8 @@ public function offsetExists($offset) : bool
      * @param string|int $offset The offset to retrieve.
      *
      * @return mixed
+     *
+     * @phpstan-return ?T
```

Comment:
I need to read up why this can't go in a regular `@return` .. I would expect it in the `@return` :thinking:

Reply:
I'm actually not sure why I did it that way. I'm looking into it

Reply:
I think this is not nullable, but since a lot of methods in the descriptors are returning null. And we had a bug where we needed to filter those, it could be that phpstan is applying some reversed logic here.

Reply:
I do remember now why I use specific phpstan tags :D

```
----------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
  82 | ERROR | [x] Method \phpDocumentor\Descriptor\Collection::get()
     |       |     does not have return type hint for its return
     |       |     value but it should be possible to add it based on
     |       |     @return annotation "?T".
 132 | ERROR | [x] Method
     |       |     \phpDocumentor\Descriptor\Collection::__get() does
     |       |     not have return type hint for its return value but
     |       |     it should be possible to add it based on @return
     |       |     annotation "?T".
 154 | ERROR | [x] Method
     |       |     \phpDocumentor\Descriptor\Collection::offsetGet()
     |       |     does not have return type hint for its return
     |       |     value but it should be possible to add it based on
     |       |     @return annotation "?T".
----------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

CS checks will kill me!

Reply:
Ah crap; we may need to report this to phpcs and/or the people who made these specific sniffs

Reply:
It's possible this is fixed already. It's kind of the same issue than parsing generics, It could have been fixed in a newer version of sniffs.

I'll probably try back later if the QA tools are updated.